sync/syncer.go

```diff
@@ -130,6 +131,7 @@ func NewSync(srv service.Service, layers *mesh.Mesh, txpool MemPool, atxpool Mem
 		txpool:         txpool,
 		atxpool:        atxpool,
 		startLock:      0,
+		currentLayer:   currentLayer,
```

Comment:
why is this needed ? we are getting the current layer from the clock we subscribed to 
we shouldn't have to add this

Reply:
if the issue is the time until the next tick we should receive the clock to the contractor and then set the logic and the subscription in here

Reply:
I'll add a technical debt, it seems that tests are failing because some assumptions regarding the channels the syncers are listening to are not correct, we need to revise the tests before making this change

Reply:
#1043